### PR TITLE
Align terminology for subflow, flow and application subflow

### PR DESCRIPTION
**Pull request by theomarzaki**
Changes to address #71 

Please Double Check: 
* logic for example mp dccp usage scenario (283)
* logic for Address ID handling (581)

### CODE REVIEW

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -205,14 +205,14 @@ data, and termination of sessions.
 DCCP, as stated in {{RFC4340}} does not provide reliable and ordered
 delivery. Consequently, multiple application subflows may be multiplexed over a
 single DCCP connection with no inherent performance penalty 
-for flows that do not require in-ordered delivery. DCCP
+for said subflows that do not require in-ordered delivery. DCCP
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:208]: Perhaps "application subflows" instead?

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -578,13 +578,13 @@ MP_HMAC MUST be set when using MP_JOIN to provide authentication (See
 MP_HMAC for details). Also MP_KEY MUST be set to provide key material
 for authentication purposes.
 
-The Address IDs of the subflow used in the initial DCCP Request/Response exchange of
-the first subflow in the connection are implicit, and have the value
+The Address IDs of the flow used in the initial DCCP Request/Response exchange
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:581]: Why is this not "subflow"? From above I thought you had gotten rid of "flow"

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 3]
```diff
@@ -1092,7 +1092,7 @@ following key security requirements to be fulfilled by MP-DCCP:
 In order to achieve these goals, MP-DCCP includes a hash-based handshake
 algorithm documented in Sections {{MP_KEY}} and {{handshaking}}. The
 security of the MP-DCCP connection depends on the use of keys that are
-shared once at the start of the first subflow and are never sent again
+shared once at the start of the first flow and are never sent again
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1095]: Why is this not "subflow"?

**Review by abrunstrom — CHANGES_REQUESTED** [review]
Thanks for the PR, looking better, but I am still a bit confused about the use of flow vs. subflow?

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 1 above)
Comment (theomarzaki) [reply draft-ietf-tsvwg-multipath-dccp.mkd:208]: i wanted to avoid repetition. But i changed it to application sub-flows.

**Comment by theomarzaki** [comment]
> Thanks for the PR, looking better, but I am still a bit confused about the use of flow vs. subflow?

  From what i understood in the definitions on line 211 / 254  and example mp-dccp usage diagram, the initial mpdccp connection is a flow, if there is a capability of adding multiple paths, then all subsequent flows become sub-flows. If the one or more sub-flows exist, the initial flow is promoted to sub-flow.

**Review by markusa — APPROVED** [review]

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 4]
```diff
@@ -957,13 +958,13 @@ mechanisms (see {{ref-mp-option-confirm}}).
...
+On reception of a first DCCP-CloseReq carrying a MP_CLOSE with valid Key Data, or due to a local decision, all subflows transition to the CLOSING state before transmitting a DCCP-Close carrying MP_CLOSE. In this case, the MP-DCCP connection socket on the host sending the DCCP-Close reflects the state of the initial subflow used during handshake with MP_KEY option. If the initial subflow no longer exists, the state moves immediately to CLOSED.
 
-Upon reception of the first DCCP-Close carrying a MP_CLOSE with valid Key Data at the peer host, all subflows, including the application socket, move to the CLOSED state. After this, a DCCP-Reset with Reset Code 1 MUST be sent on any subflow in response to a received DCCP-Close containing a valid MP_CLOSE option.
+Upon reception of the first DCCP-Close carrying a MP_CLOSE with valid Key Data at the peer host, all subflows, including the MP-DCCP connection socket, move to the CLOSED state. After this, a DCCP-Reset with Reset Code 1 MUST be sent on any subflow in response to a received DCCP-Close containing a valid MP_CLOSE option.
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:965]: ... as well as the MP-DCCP connection socket ...

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 5]
```diff
@@ -254,11 +254,12 @@ this context by a 4-tuple of source and destination address/ port pairs.
 
 Subflow: A flow of DCCP segments operating over an individual path,
 which forms part of a larger MP-DCCP connection. A subflow is started
-and terminated similar to a regular (single-path) DCCP connection.
+and terminated similar to a regular (single-path) DCCP connection. The term subflow
+can be used for one or more paths in an MP-DCCP connection.
```
[the remark this answers is not in the document]
Comment (theomarzaki) [reply draft-ietf-tsvwg-multipath-dccp.mkd:258]: Perhaps a better wording would be better,

" The term subflow can also be used to refer to an MP-DCCP connection with one path."

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 6]
```diff
@@ -254,11 +254,12 @@ this context by a 4-tuple of source and destination address/ port pairs.
...
 Subflow: A flow of DCCP segments operating over an individual path,
 which forms part of a larger MP-DCCP connection. A subflow is started
-and terminated similar to a regular (single-path) DCCP connection.
+and terminated similar to a regular (single-path) DCCP connection. The term subflow
+can also be used to refer to an MP-DCCP connection with a single path.
 
 (MP-DCCP) Connection: A set of one or more subflows, over which an
 application can communicate between two hosts. There is a one-to-one
-mapping between a connection and an application socket.
+mapping between a connection and an MP-DCCP connection socket.
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:262]: This does not read so well. Also not sure the MP-DCCP features need to be/are exposed to the application? Possibly this should be something like "The MP-DCCP connection is exposed as a single DCCP socket to the application." ?

But possibly we can leave it for now and open a new issue.

**Review by abrunstrom — APPROVED** [review]
Few last comments. After fixing them I think we can merge.

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 6 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Incorporated https://github.com/markusa/ietf-multipath-dccp/pull/97/commits/d86db59cedaeefde38311dd6846ae93625dd9cb8

**Pull request merged**